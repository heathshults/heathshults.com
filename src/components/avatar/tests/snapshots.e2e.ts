import { snapIt } from '../../../test';

const component = 'avatar';

describe(component, () => {
  const snap = snapIt(component);

  describe('renders', () => {
    snap('nothing if no attributes are set', '<hs-avatar></hs-avatar>');
    snap('text inside', '<hs-avatar text="GP"></hs-avatar>');
    snap('correctly sized circles', '<hs-avatar size="super" text="GP"></hs-avatar>');
    snap('an image', '<hs-avatar src="my-test-image.png"></hs-avatar>');
    snap('2 images', '<hs-avatar src="my-test-image.png" src2="my-second-test-image.png"></hs-avatar>');
    snap(
      'images, text and sizes it correctly',
      '<hs-avatar src="my-test-image.png" src2="my-second-test-image.png" size="super" text="GP"></hs-avatar>'
    );
  });
});
