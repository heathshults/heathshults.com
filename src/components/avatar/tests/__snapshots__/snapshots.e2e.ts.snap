// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`avatar renders 2 images 1`] = `"<hs-avatar src=\\"my-test-image.png\\" src2=\\"my-second-test-image.png\\" class=\\"hydrated\\"><div class=\\"hs-avatar\\"><img class=\\"hs-avatar__img\\" src=\\"my-test-image.png\\"></div></hs-avatar>"`;

exports[`avatar renders an image 1`] = `"<hs-avatar src=\\"my-test-image.png\\" class=\\"hydrated\\"><div class=\\"hs-avatar\\"><img class=\\"hs-avatar__img\\" src=\\"my-test-image.png\\"></div></hs-avatar>"`;

exports[`avatar renders correctly sized circles 1`] = `"<hs-avatar size=\\"super\\" text=\\"GP\\" class=\\"hydrated\\"><div class=\\"hs-avatar u-super\\" data-text=\\"GP\\"></div></hs-avatar>"`;

exports[`avatar renders images, text and sizes it correctly 1`] = `"<hs-avatar src=\\"my-test-image.png\\" src2=\\"my-second-test-image.png\\" size=\\"super\\" text=\\"GP\\" class=\\"hydrated\\"><div class=\\"hs-avatar u-super\\" data-text=\\"GP\\"><img class=\\"hs-avatar__img\\" src=\\"my-test-image.png\\"></div></hs-avatar>"`;

exports[`avatar renders nothing if no attributes are set 1`] = `"<hs-avatar class=\\"hydrated\\"></hs-avatar>"`;

exports[`avatar renders text inside 1`] = `"<hs-avatar text=\\"GP\\" class=\\"hydrated\\"><div class=\\"hs-avatar\\" data-text=\\"GP\\"></div></hs-avatar>"`;
